### PR TITLE
feat: add 2 more chunk states for writing chunks to object store

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #

Add 2 more ChunkState to support writing chunks to object stores. Second PR for #1081

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: server/src/db.rs [hunk 1]
```diff
@@ -409,6 +409,9 @@ impl Db {
         partition_key: &str,
         chunk_id: u32,
     ) -> Result<Arc<DBChunk>> {
+        // TODO: the below was done for MB --> Parquet. Need to rewrite to RB
```
Comment (NGA-TRAN) [inline server/src/db.rs:412]: FYI: The previous merge #1097 created a framework to write MB --> Parquet. Now we decided to write RB --> Parquet instead. I am trying to do it but it takes more time. So the changes in this db.rs file is just something I try to see how to do it, not done yet. You do not need to review it yet. It compiles but not run yet. Please help me review the other 2 files.

File: server/src/db/catalog/chunk.rs [hunk 2]
```diff
@@ -31,10 +31,14 @@ pub enum ChunkState {
     Moving(Arc<MBChunk>),
 
     /// Chunk has been completely loaded in the read buffer
-    Moved(Arc<ReadBufferDb>), // todo use read buffer chunk here
+    /// Still need both MB and RD for moving the MB to Object Store
```
Comment (alamb) [inline server/src/db/catalog/chunk.rs:34]: I don't understand this comment about needing both MB and RD

File: server/src/db/catalog/chunk.rs [hunk 3]
```diff
@@ -31,10 +31,14 @@ pub enum ChunkState {
...
+    Moved(Arc<ReadBufferDb>), // todo use read buffer chunk instead of ReadBufferDb
 
-    /// Chunk has been completely loaded in the object store
-    ObjectStore(Arc<ParquetChunk>),
+    // Chunk is actively writing to object store
+    WritingToObjectStore(Arc<ReadBufferDb>), // todo use read buffer chunk instead of ReadBufferD
+
+    // Chunk has been completely written into object store
+    WrittenToObjectStore(Arc<ReadBufferDb>, Arc<ParquetChunk>),
```
Comment (alamb) [inline server/src/db/catalog/chunk.rs:41]: I eventually imagine that we'll have states like:

```
// in read buffer, Persisted in object store
PersistedCachedMem(RBChunk), 

// persisted to object store, NOT in read buffer,  but available for querying locally in parquet format
PersistedCachedLocally(ParquetChunk), 

// persisted in object store, but not available for query yet (needs to be fetched)
Persisted(ParquetChunk), 
```

**Review by alamb — APPROVED** [review]
looks like another good step in the right direction in my opinion. 

Eventually we should get some tests hooked up but I think that can happen as a follow on PR

File: server/src/db/catalog/chunk.rs (hunk 2 above)
Comment (NGA-TRAN) [reply server/src/db/catalog/chunk.rs:34]: Ah, this was the old one to write from MB. Will remove. Good catch :)

**Pull request merged**